optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/util/configuration/security/ResponseHeadersConfiguration.java

```diff
@@ -112,4 +117,35 @@ public String toString() {
         + getContentSecurityPolicy()
         + ")";
   }
+
+  public Map<String, String> getHeadersWithValues() {
+    final Map<String, String> headers = new HashMap<>();
+
+    final List<String> strictTransportSecurityHeaderValues = new ArrayList<>();
+    if (getHttpStrictTransportSecurityMaxAge() > 0) {
+      strictTransportSecurityHeaderValues.add("max-age=" + getHttpStrictTransportSecurityMaxAge());
+    }
+    if (getHttpStrictTransportSecurityIncludeSubdomains()) {
+      strictTransportSecurityHeaderValues.add("includeSubDomains");
+    }
+    if (!strictTransportSecurityHeaderValues.isEmpty()) {
+      headers.put(
+          "Strict-Transport-Security",
+          String.join(HEADER_DELIMITER, strictTransportSecurityHeaderValues));
+    }
+
+    if (getXsssProtection() != null && getXsssProtection().length() > 0) {
+      headers.put("X-XSS-Protection", getXsssProtection());
+    }
+
+    if (getXContentTypeOptions()) {
+      headers.put("X-Content-Type-Options", "nosniff");
+    }
+
+    if (getContentSecurityPolicy() != null && getContentSecurityPolicy().length() > 0) {
+      headers.put("Content-Security-Policy", getContentSecurityPolicy());
+    }
```

Comment:
Instead of using a string for the header properties, you can use existing contants

For example, `HttpHeaders.X_XSS_PROTECTION` instead of `"X-XSS-Protection"`